### PR TITLE
e2e (nextjs-v12 sample)

### PR DESCRIPTION
**Pull request by Andranik-mikhaielian**
Screenshot 2023-02-06 at 10 39 17 PM

### CODE REVIEW

**Comment by PeaceDate** [comment]
You've missed `vite-react-simple/e2e/tests/commonChecks.cy.ts` and `vite-react-simple/e2e/tests/viteAppChecks.cy.ts` during refactor of `openLocalhost()` method

File: cypress/fixtures/constants.ts [hunk 1]
```diff
@@ -702,6 +702,34 @@ export class Constants {
...
+                buttonName: 'Federated Catch All',
+                header: 'PDP!!!'
+            },
+            checkoutButton : {
+                buttonName: 'Checkout',
+                header: 'checkout page',
+                subheader: 'This is a federated page owned by localhost:3000'
+            },
+            zeitButton: 'ZEIT',
+            gitButton: 'GitHub'
```
Comment (PeaceDate) [inline cypress/fixtures/constants.ts]: We already have these constants. Please check, what we have and use it

File: nextjs-v12/e2e/checkHomeApp.cy.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        federatedButton: string,
+        federatedHeader: string,
+        checkoutButton: string,
+        checkoutHeader: string,
+        checkoutSubheader: string,
+        zeitButton: string,
+        gitButton: string,
+        host: number
+    }
+) => {
```
Comment (PeaceDate) [inline nextjs-v12/e2e/checkHomeApp.cy.ts]: You can get rid of this duplicate constants.

**Review by PeaceDate — CHANGES_REQUESTED** [review]

**Review by PeaceDate — APPROVED** [review]

File: cypress/common/base.ts [hunk 3]
```diff
@@ -749,11 +751,20 @@ export class BaseMethods {
...
+        path,
+        wait=0
+    }: {
+        number: number,
+        path?: string,
+        wait?: number
+    }): Cypress.Chainable<Cypress.AUTWindow> {
         return path ?
-            cy.visit(`${Cypress.env(`localhost${number}`)}/${path}`)
+            cy.visit(`${Cypress.env(`localhost${number}`)}/${path}`).wait(wait)
```
Comment (edchai) [inline cypress/common/base.ts]: do not like the idea of wait with visit. Try use waitUntil with param something like "isWithWait" or just move wait to spec level

File: cypress/fixtures/constants.ts [hunk 4]
```diff
@@ -702,6 +702,12 @@ export class Constants {
                 'Learn React',
                 'Vite Docs'
             ]
+        },
+        nextJSv12: {
```
Comment (edchai) [inline cypress/fixtures/constants.ts]: nextJSv12App

File: cypress/fixtures/constants.ts [hunk 5]
```diff
@@ -702,6 +702,12 @@ export class Constants {
...
+            federatedButton: {
+                buttonName: 'Federated Catch All',
```
Comment (edchai) [inline cypress/fixtures/constants.ts]: name

File: nextjs-v12/e2e/checkNextJSv12Apps.cy.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
+import {BaseMethods} from "../../cypress/common/base";
+import {baseSelectors} from "../../cypress/common/selectors";
+import {Constants} from "../../cypress/fixtures/constants";
+const basePage: BaseMethods = new BaseMethods()
+
+const appsData = [
+    {
+        appName: 'Checkout',
```
Comment (edchai) [inline nextjs-v12/e2e/checkNextJSv12Apps.cy.ts]: all appNames to constants

File: nextjs-v12/e2e/checkNextJSv12Apps.cy.ts [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+) => {
+    describe(`Check ${property.appName} app elements and links are visible and not broken`, () => {
+        beforeEach(() => {
+            basePage.openLocalhost({
+                number: property.host,
+                wait: 2000
+            })
+        })
+
+        it (`Check application functionality and elements`, () => {
```
Comment (edchai) [inline nextjs-v12/e2e/checkNextJSv12Apps.cy.ts]: have no idea what is going on here. Better to split to multiple its with proper names

File: nextjs-v12/package.json [hunk 8]
```diff
@@ -15,5 +15,8 @@
   },
   "dependencies": {
     "concurrently": "^7.0.0"
+  },
+  "devDependencies": {
```
Comment (edchai) [inline nextjs-v12/package.json]: as I understand we do not push major dependencies

File: angular11-microfrontends-scully/e2e/checkAngular11SkullyApps.cy.ts [hunk 9]
```diff
@@ -507,7 +513,10 @@ describe('Check Product Application', () => {
     const productList = ProductList
 
     beforeEach(() => {
-        basePage.openLocalhost(4200, '/product')
+        basePage.openLocalhost({
+            number: 4200,
+            path: '/product'
```
Comment (edchai) [inline angular11-microfrontends-scully/e2e/checkAngular11SkullyApps.cy.ts]: to constants -> hrefs

**Review by edchai — CHANGES_REQUESTED** [review]

File: cypress/fixtures/constants.ts [hunk 10]
```diff
@@ -985,6 +991,9 @@ export class Constants {
         thirdPartyScriptsPostRequestPath: 'https://www.google-analytics.com/j/collect?**',
         viteReactSimpleApp: {
             viteLink: 'https://vitejs.dev/guide/features.html'
+        },
+        commonAngularData: {
```
Comment (edchai) [inline cypress/fixtures/constants.ts]: strange place. If this link is used in only one place, this is not an app name as it supposed to be, if it used in multiple places -> move it to proper location inside commonConstantsData

**Review by edchai — CHANGES_REQUESTED** [review]

File: cypress/fixtures/constants.ts (hunk 10 above)
Comment (Andranik-mikhaielian) [reply cypress/fixtures/constants.ts]: It is used in a single place, so I renamed it in the constants.ts

**Review by edchai — APPROVED** [review]

**Comment by PeaceDate** [comment]
Closed and reopen as a new PR ---> #2796

**Comment by PeaceDate** [comment]
Closed and reopen as a new PR ---> https://github.com/module-federation/module-federation-examples/pull/2796

**Pull request closed without merging**